nautobot/extras/utils.py

```diff
@@ -129,6 +130,17 @@ def get_query(self):
         """
         Given an extras feature, return a Q object for content type lookup
         """
+
+        # The `populate_model_features_registry` function is called in the `FeatureQuery().get_query` method instead of
+        # `ExtrasConfig.ready` because `FeatureQuery().get_query` is called before `ExtrasConfig.ready`.
+        # This is because `FeatureQuery` is a helper class used in `Forms` and `Serializers` that are called during the
+        # initialization of the application, before `ExtrasConfig.ready` is called.
+        # Calling `populate_model_features_registry` in `ExtrasConfig.ready` would lead to an outdated `model_features`
+        # `registry` record being used by `FeatureQuery`.
+        # As the `populate_model_features_registry` can be resource-intensive. The check the conditional check is used to
+        # avoid calling the function multiple times and optimize the performance of the application.
+        if not registry["model_features"].get("relationships"):
+            populate_model_features_registry()
```

Comment:
I gotta say I don't like this. Should we instead be deferring the evaluation of `FeatureQuery()` in some fashion? Why wasn't this an issue before this PR?

Reply:
This was not an issue because the models `@extras_features` are evaluated first before serializers and forms `FeatureQuery().get_query()`.

Reply:
Any ideas on how I can archive delay the evaluation of `FeatureQuery()`?

Reply:
Half-baked idea that came to me just now - what if instead of having a separate `populate_model_features_registry()` that we need to call directly after all model classes are loaded, what if we added logic to `BaseModel` (probably in something like `__init_subclass__()` ??) to where the model automatically adds itself to the appropriate model_features in the registry when it's declared/loaded?